yarn-project/foundation/src/config/secret_value.ts

```diff
@@ -0,0 +1,41 @@
+import { type ZodType } from 'zod';
+
+/**
+ * A class wrapping a secret value to protect it from accidently being leaked in logs
+ */
+export class SecretValue<T> {
+  /** The secret value */
+  #value: T;
+
+  constructor(value: T, private readonly redactedValue = '[Redacted]') {
+    this.#value = value;
+  }
+
+  /**
+   * Returns the wrapped value
+   */
+  public getValue(): T {
+    return this.#value;
+  }
+
+  /**
+   * Returns a redacted string representation of the value
+   */
+  public toString(): string {
+    return this.redactedValue;
+  }
+
+  /**
+   * Returns a redacted string representation of the value
+   */
+  public toJSON(): string {
+    return this.redactedValue;
+  }
```

Comment:
I was going to point out that this breaks config serialization/deserialization, which is currently used eg in the snapshot manager. But I also got a ticket to delete the snapshot manager, so it's fine.